### PR TITLE
feat(bigtable): Accept nil gc_rule arguments for column_family create/update

### LINKED ISSUE #3680: Cloud Bigtable: Unable to create column family without a garbage collection policy
I tried creating a table with a column family that had no garbage collection policy (which would be default,) but I receive an error when it runs.

```
    table = bigtable.create_table @instance_id, table_id
    table.column_family("stats_summary").create()
```

```
  table = bigtable.create_table instance_id, table_id do |column_families|
    column_families.add(column_family)
  end

```
These both fail.

### PR DESCRIPTION
**Pull request by quartzmo**
[closes #3680]

### CODE REVIEW

File: google-cloud-bigtable/acceptance/bigtable/table_test.rb [hunk 1]
```diff
@@ -78,7 +78,7 @@
 
     table = bigtable.create_table(instance_id, table_id) do |cfs|
       cfs.add("cf1", Google::Cloud::Bigtable::GcRule.max_versions(1))
```
Comment (billyjacobson) [inline google-cloud-bigtable/acceptance/bigtable/table_test.rb]: Could you have cf1 be the one without the rule and cf2 have max versions? Seems more clear to have the default one be first

Comment (quartzmo) [reply google-cloud-bigtable/acceptance/bigtable/table_test.rb]: Done

**Comment by quartzmo** [comment]
@blowmage Can you review when you get the chance, thanks!

File: google-cloud-bigtable/lib/google/cloud/bigtable/table.rb [hunk 2]
```diff
@@ -238,8 +238,8 @@ def exists?
         # update, or delete operations.
         #
         # @param name [String] Name of the column family
-        # @param gc_rule [Google::Cloud::Bigtable::GcRule] Optional.
-        #   Required for create and update operations.
+        # @param gc_rule [Google::Cloud::Bigtable::GcRule, nil] A garbage
```
Comment (blowmage) [inline google-cloud-bigtable/lib/google/cloud/bigtable/table.rb]: I think listing `nil` here might be confusing. What do you think about something like this:

```ruby
# @param gc_rule [Google::Cloud::Bigtable::GcRule] The garbage collection
#   rule to be used for the column family. Optional. The service default value
#   will be used when not specified.
```

Comment (quartzmo) [reply google-cloud-bigtable/lib/google/cloud/bigtable/table.rb]: I like it. I'll update with that wording everywhere.

Comment (quartzmo) [reply google-cloud-bigtable/lib/google/cloud/bigtable/table.rb]: Updated.

**Pull request merged**